core/src/components/fab-button/fab-button.tsx

```diff
@@ -153,7 +153,7 @@ export class FabButton implements ComponentInterface, AnchorInterface, ButtonInt
         onClick={(ev: Event) => openURL(this.href, ev, this.routerDirection)}
       >
         <span class="close-icon">
-          <ion-icon name="close" lazy={false}></ion-icon>
+          <slot name="close"></slot>
```

Comment:
Why not use the previous `ion-icon` as default? Without it, you're forcing people to change their code on the next upgrade.

Reply:
Watching for changes to detect if a slot if provided in order to not display a default icon seems an overhead to me

Furthermore, I think the the component should be "ionicons reference free" respectively it should not be mandatory to fetch the ionicons in order to use that component

But fill free to improve the PR

Reply:
> Watching for changes to detect if a slot if provided

You don't have to watch for changes. Anything inside the `slot` declaration will be rendered if no `slot` is provided. I agree with the "reference free" but to me it seems unnecessary to introduce a breaking change.

I had created a [PR](https://github.com/ionic-team/ionic/pull/18519) actually but I think yours is good because it has the documentation changes as well.